panther_analysis_tool/main.py

```diff
@@ -179,17 +180,22 @@ def load_module(filename: str) -> Tuple[Any, Any]:
     return module, None
 
 
-def load_analysis_specs(directories: List[str]) -> Iterator[Tuple[str, str, Any, Any]]:
+def load_analysis_specs(directories: List[str],
+                        ignored_files: OPT[list] = None) ->\
+        Iterator[Tuple[str, str, Any, Any]]:
     """Loads the analysis specifications from a file.
 
     Args:
         directories: The relative path to Panther policies or rules.
+        ignored_files: Files that Panther Analysis Tool should not process
 
     Yields:
         A tuple of the relative filepath, directory name, and loaded analysis specification dict.
     """
     # setup a list of paths to ensure we do not import the same files
     # multiple times, which can happen when testing from root directory without filters
+    if ignored_files is None:
+        ignored_files = []
```

Comment:
if you add a `default=""` to the parser... 1. would it work? 2. could we get rid of this?

Reply:
It seems no. I tried the default as an empty string and an empty list and both still ended up being `None` without this snippet. It seems the only way to drop this is to make the default argument an empty list which goes against the "no mutable defaults" standard.

Reply:
You can add an [empty list as default in the parser argument definition](https://github.com/panther-labs/panther_analysis_tool/pull/139/files#diff-4c3406e46ca13890fa64d295f86ed01e0b0bcd3d28031f1fc97978bb8fca2da3R1284) and make `ignored_files` a required argument for this function.

Reply:
🤦🏼  I am not sure why that didn't occur to me. Changed accordingly.